Flaky test: sgld

```
======================================================================

FAIL: test_operator_gpu.test_sgld

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/usr/local/lib/python3.5/dist-packages/nose/case.py", line 198, in runTest

    self.test(*self.arg)

  File "/usr/local/lib/python3.5/dist-packages/nose/util.py", line 620, in newfunc

    return func(*arg, **kw)

  File "/work/mxnet/tests/python/gpu/../unittest/common.py", line 173, in test_new

    orig_test(*args, **kwargs)

  File "/work/mxnet/tests/python/gpu/../unittest/test_optimizer.py", line 520, in test_sgld

    compare_optimizer_noise_seeded(opt1(**kwarg), opt2(**kwarg), shape, dtype, seed)

  File "/work/mxnet/tests/python/gpu/../unittest/test_optimizer.py", line 511, in compare_optimizer_noise_seeded

    assert_almost_equal(w1.asnumpy(), w2.asnumpy(), rtol=rtol, atol=atol)

  File "/work/mxnet/python/mxnet/test_utils.py", line 495, in assert_almost_equal

    raise AssertionError(msg)

AssertionError: 

Items are not equal:

Error 5.851562 exceeds tolerance rtol=0.000100, atol=0.000010.  Location of maximum error:(2, 2, 0), a=0.734863, b=0.734375

 a: array([[[ 0.628   ,  0.6836  ,  0.2057  ,  0.909   ,  0.3188  ],

        [ 0.2113  ,  0.3286  ,  0.8438  ,  0.847   ,  0.5557  ],

        [ 0.6147  ,  0.595   ,  0.858   ,  0.0252  ,  0.1906  ],...

 b: array([[[ 0.628   ,  0.6836  ,  0.2057  ,  0.909   ,  0.3188  ],

        [ 0.2113  ,  0.3286  ,  0.8438  ,  0.847   ,  0.5557  ],

        [ 0.6147  ,  0.595   ,  0.858   ,  0.0252  ,  0.1906  ],...

-------------------- >> begin captured logging << --------------------

common: INFO: Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1061286115 to reproduce.

--------------------- >> end captured logging << ---------------------

```
occured in http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Funix-gpu/activity for PR https://github.com/apache/incubator-mxnet/pull/14237

--- a/tests/python/unittest/test_optimizer.py
+++ b/tests/python/unittest/test_optimizer.py
@@ -461,7 +461,6 @@ class PySGLD(mx.optimizer.Optimizer):
 
 
 @with_seed()
-@unittest.skip("test fails intermittently. temporarily disabled till it gets fixed. tracked at https://github.com/apache/incubator-mxnet/issues/14241")
 def test_sgld():
     opt1 = PySGLD
     opt2 = mx.optimizer.SGLD
@@ -518,7 +517,9 @@ def test_sgld():
                 if (dtype == np.float16 and ('multi_precision' not in kwarg or
                     not kwarg['multi_precision'])):
                     continue
-                compare_optimizer_noise_seeded(opt1(**kwarg), opt2(**kwarg), shape, dtype, seed)
+                atol = 1e-2 if dtype == np.float16 else 1e-3
+                rtol = 1e-4 if dtype == np.float16 else 1e-5
+                compare_optimizer_noise_seeded(opt1(**kwarg), opt2(**kwarg), shape, dtype, seed, atol=atol, rtol=rtol)
 
 
 

